wgpu-hal/src/gles/device.rs

```diff
@@ -402,11 +473,35 @@ impl crate::Device<super::Api> for super::Device {
         let gl = &self.shared.context.lock();
         gl.bind_buffer(buffer.target, Some(buffer.raw));
         for range in ranges {
-            gl.flush_mapped_buffer_range(
-                buffer.target,
-                range.start as i32,
-                (range.end - range.start) as i32,
-            );
+            if self
```

Comment:
similarly, let's just check `emulate_map_allocation` since we have to check it anyway